examples/rust/examples/maker/maker.rs

```diff
@@ -38,100 +41,117 @@ const TOS_ACCEPTANCE: &str = "I accept the Valorem Terms of Service at https://a
+    } else if node_endpoint.starts_with("ws") {
+        let provider = match Ws::connect(node_endpoint).await {
+            Ok(connection) => {
+                // Decrease the ethers polling time from the default of 7 seconds to 1 second.
+                Provider::<Ws>::new(connection).interval(Duration::from_secs(1))
+            }
+            Err(connection_error) => {
+                // If we cannot connect to our provider, we are in trouble. Panic out instead of
+                // trying to implement reconnection logic as most of the time the node will be local.
+                panic!("Websocket connection error: {connection_error:?}")
+            }
         };
 
-        op.retry(
-            &ExponentialBuilder::default()
-                .with_jitter()
-                .with_max_times(10),
-        )
-        .await?;
+        EthersProvider::WsProvider(provider)
     } else {
-        // IPC
-        let op = || async {
-            warn!("Re/starting maker");
-            run(
-                Arc::new(Provider::connect_ipc(settings.node_endpoint.clone()).await?),
-                settings.clone(),
-            )
-            .await
+        let provider = match Provider::connect_ipc(node_endpoint).await {
+            Ok(connection) => {
+                // Decrease the ethers polling time from the default of 7 seconds to 1 second.
+                connection.interval(Duration::from_secs(1))
+            }
+            Err(connection_error) => {
+                // If we cannot connect to our local provider, we are in trouble just panic out instead.
+                panic!("IPC connection error: {connection_error:?}")
+            }
         };
 
-        op.retry(
-            &ExponentialBuilder::default()
-                .with_jitter()
-                .with_max_times(10),
-        )
-        .await?;
+        EthersProvider::IpcProvider(provider)
     }
-
-    Ok(())
 }
 
 // Main execution function. This is not expected to return.
 async fn run<P: JsonRpcClient + 'static>(
     provider: Arc<Provider<P>>,
     settings: Settings,
-) -> Result<(), anyhow::Error> {
+) -> Option<()> {
     let session_cookie = setup(
         settings.valorem_endpoint.clone(),
         settings.wallet.clone(),
         settings.tls_config.clone(),
         &provider,
     )
-    .await;
+    .await?;
 
     // Now there is a valid authenticated session, connect to the RFQ stream
     let mut client = RfqClient::with_interceptor(
         Channel::builder(settings.valorem_endpoint.clone())
-            .tls_config(settings.tls_config.clone())?
+            .tls_config(settings.tls_config.clone())
+            .ok()?
             .http2_keep_alive_interval(Duration::new(75, 0))
             .keep_alive_timeout(Duration::new(10, 0))
             .timeout(Duration::from_secs(10))
             .connect_timeout(Duration::from_secs(10))
             .connect()
-            .await?,
+            .await
+            .ok()?,
```

Comment:
be nice if all errors which cause restart are logged rather than silent

Reply:
Done